python/lsst/ip/isr/calibType.py

```diff
@@ -0,0 +1,630 @@
+            if getattr(self, attr) != getattr(other, attr):
+                return False
+
+        return True
+
+    def getMetadata(self):
+
+        """Retrieve metadata associated with this calibration.
+
+        Returns
+        -------
+        meta : `lsst.daf.base.PropertyList`
+            Metadata. The returned `~lsst.daf.base.PropertyList` can be
+            modified by the caller and the changes will be written to
+            external files.
+        """
+        return self._metadata
+
+    def setMetadata(self, metadata):
+        """Store a copy of the supplied metadata with this calibration.
+
+        Parameters
+        ----------
+        metadata : `lsst.daf.base.PropertyList`
+            Metadata to associate with the calibration.  Will be copied and
+            overwrite existing metadata.
+        """
+        if metadata:
+            self._metadata = copy.copy(metadata)
+
+        # Ensure that we have the obs type required by calibration ingest
+        self._metadata["OBSTYPE"] = self._OBSTYPE
+        self._metadata[self._OBSTYPE + "_SCHEMA"] = self._SCHEMA
+        self._metadata[self._OBSTYPE + "_VERSION"] = self._VERSION
+
+    def updateMetadata(self, *args, setDate=False, **kwargs):
+        """Update metadata keywords with new values.
+
+        Parameters
+        ----------
+        setDate : `bool`, optional
+            Ensure the metadata CALIBDATE fields are set to the current datetime.
+        kwargs :
+            Set of key=value pairs to assign to the metadata.
+        """
+        mdOriginal = self.getMetadata()
+        mdSupplemental = dict()
+
+        self._metadata["DETECTOR"] = self._detectorName
+        self._metadata["DETECTOR_SERIAL"] = self._detectorSerial
+
+        if setDate:
+            date = datetime.datetime.now()
+            mdSupplemental['CALIBDATE'] = date.isoformat()
+            mdSupplemental['CALIB_CREATION_DATE'] = date.date().isoformat()
+            mdSupplemental['CALIB_CREATION_TIME'] = date.time().isoformat()
+
+        for key, value in kwargs.items():
+            mdSupplemental[key] = value
+
+        mdOriginal.update(mdSupplemental)
+
+    @classmethod
+    def readText(cls, filename):
+        """Read calibration representation from a yaml/ecsv file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file containing the calibration definition.
+
+        Returns
+        -------
+        calib : `~lsst.ip.isr.IsrCalibType`
+            Calibration class.
+        """
+        data = ''
+        try:
+            data = Table.read(filename, format='ascii.ecsv')
+        except Exception as e1:
+            try:
+                with open(filename, 'r') as f:
+                    data = yaml.load(f, Loader=yaml.CLoader)
+            except Exception as e2:
+                raise e2(f"Failed to read as ecsv as well: {e1}")
+
+        if isinstance(data, dict):
```

Comment:
I don't think we need to be doing instance checks here.  I think that we can safely say:

```
if endswith yaml:
    read yaml
    pass yaml to cls.fromDict 
elif enswith ecsv:
    read table
    process table
else:
    get upset
```